src/buffers/disk/leveldb_buffer.rs

```diff
@@ -64,45 +64,88 @@ unsafe impl Send for Writer {}
 impl Clone for Writer {
     fn clone(&self) -> Self {
         Self {
-            db: Arc::clone(&self.db),
+            db: self.db.as_ref().map(Arc::clone),
             offset: Arc::clone(&self.offset),
             write_notifier: Arc::clone(&self.write_notifier),
             blocked_write_tasks: Arc::clone(&self.blocked_write_tasks),
             writebatch: Writebatch::new(),
             batch_size: 0,
             max_size: self.max_size,
             current_size: Arc::clone(&self.current_size),
+            slot: None,
         }
     }
 }
 
-impl Sink for Writer {
-    type SinkItem = Event;
-    type SinkError = ();
+impl Sink<Event> for Writer {
+    type Error = ();
+
+    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        if self.slot.is_none() {
+            Poll::Ready(Ok(()))
+        } else {
+            self.poll_flush(cx)
```

Comment:
If we flush in `poll_ready`, doesn't that effectively mean that we never buffer/batch any events to be sent? That could be an explanation for the reduced write throughputs.

Reply:
We flush only if slot isn't empty which also means that we failed sending the slotted event which means the disk buffer was full. But you were close, the issue seems to be that I updated the benches to flush for each event, I'm currently testing if that was the main issue.